java/src/main/native/src/NvcompJni.cpp

```diff
@@ -119,30 +120,36 @@ Java_ai_rapids_cudf_nvcomp_NvcompJni_batchedLZ4CompressAsync(JNIEnv* env,
     auto out_ptrs             = reinterpret_cast<void* const*>(j_out_ptrs);
     auto compressed_out_sizes = reinterpret_cast<std::size_t*>(j_compressed_sizes_out_ptr);
     auto stream               = reinterpret_cast<cudaStream_t>(j_stream);
-    auto status               = nvcompBatchedLZ4CompressAsync(in_ptrs,
+    // FIXME how to use these statuses ? They are not used either in the corresponding
+    // decompressor.
```

Comment:
How will we track this? Follow-up work?

Reply:
I merged this as-is from https://github.com/rapidsai/cudf/pull/19309
@firestarman?

Reply:
What does the new API write into this `comp_statuses` array? Compression status for each chunk?

Reply:
Just read the doc:
>device_statuses – [out] Array with size num_chunks of statuses in device-accessible memory. This argument needs to be preallocated. For each chunk, if the compression is successful, the status will be set to `nvcompSuccess`, and an error code otherwise.


So should we launch another kernel `thrust::count_if(any error)`?

Reply:
If we don't want to sync stream here, we probably need to return the entire status array for checking later on. Otherwise, we would not be able to check for compression status of each data chunk. The `status` returned from this async API is only checking for launching the kernel, not checking the compression results.

Reply:
yes, this is pretty much what we do in libcudf
since this is unrelated to nvcomp 5, I think this could be a follow-up. Also, not sure if you want to use the adapter here long-term.